src/juju/apiHooks.ts

```diff
@@ -16,13 +18,12 @@ import { actions as jujuActions } from "store/juju";
 import { getModelByUUID, getModelUUIDFromList } from "store/juju/selectors";
 import { useAppSelector, useAppDispatch } from "store/store";
 
-export const useModelConnection = (
-  response: (
-    conn?: ConnectionWithFacades | null,
-    error?: string | null,
-  ) => void,
-  modelUUID?: string,
-) => {
+type ModelConnectionCallback = (
+  conn?: ConnectionWithFacades | null,
+  error?: string | null,
+) => void;
+
+export const useModelConnectionCallback = (modelUUID?: string) => {
```

Comment:
I've split up this hook so that you can get a function to connect to a model and call a callback.